From 62703572d091e32dd12b2eec79fed93845924847 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Fri, 02 Oct 2020 23:05:02 +0300
Subject: [PATCH] [SW-21029] ppboot: gaudi: add support for bss section

This patch adds support for bss section to ppboot
so that SW can use uninitialized global vars. Futher,
these variables will be cleared in the crt before jumping
to ppboot_main. These were required to be added
in Goya2 and hence the same is being ported to
goya/gaudi as well. It’s needed to prevent compilation issues

Change-Id: I01e249f05aa9d5c48da73e374c747fa46bd3044b
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi/ppboot.lds.S b/gaudi/ppboot.lds.S
index e078b30..1e5be88 100644
--- a/gaudi/ppboot.lds.S
+++ b/gaudi/ppboot.lds.S
@@ -20,6 +20,13 @@
 		* (.text*)
 	} >.sram
 
+	.bss : {
+		. = ALIGN(8);
+		__bss_start = . ;
+		*(.bss*)
+		__bss_end = . ;
+	} > .sram
+
 	.stack : {
 		. = ALIGN(8);
 		__stack = . ;
